docs/detectors/go.md

```diff
@@ -2,36 +2,45 @@
 
 ## Requirements
 
-Go detection depends on the following to successfully run:
+Go detection runs when one of the following files is found in the project:
 
-- `go.mod` or `go.sum` files
+- `go.mod` or `go.sum`
 
 ## Default Detection strategy
 
-Go detection depends on the following to successfully run:
+Default Go detection depends on the following to successfully run:
 
 - Go v1.11+.
 
 Full dependency graph generation is supported if Go v1.11+ is present
 on the build agent. If no Go v1.11+ is present, fallback detection
 strategy is performed.
 
-Go detection is performed by parsing output from executing [go list -m
--json all](1). To generate the graph, the command [go mod graph](2) is
-executed, this only adds edges between the components that were
-already registered by `go list`.
+Go detection is performed by parsing output from executing 
+[go list -mod=readonly -m -json all](1). To generate the graph, the command 
+[go mod graph](2) is executed. This only adds edges between the components 
+that were already registered by `go list`.
 
 ## Fallback Detection strategy
 
+The fallback detections trategy is known to overreport (see the 
+[known limitations](#known-limitations)). Read through the 
+[troubleshooting section](#troubleshooting-failures-to-run-the-default-go-detection-strategy)
+for tips on how to ensure that the newer, more accurate default 
+detection strategy runs successfully. 
+
+To force the fallback detection strategy, set the environment 
+variable: `DisableGoCliScan=true`
+
 ### `go.mod` before go 1.17
 
 Go detection is performed by parsing any `go.mod` or `go.sum` found
 under the scan directory.
 
-Only root dependency information is generated instead of full graph.
-I.e. tags the top level component or explicit dependency a given
-transitive dependency was brought by. Given a dependency tree A -> B
--> C, C's root dependency is A.
+Only root dependency information is generated in the fallback detection 
+strategy. The full graph is not detected. Given a dependency tree `A->B->C`, 
+only `A`, `C`'s root dependency, is detected.
```

Comment:
I dont think i understand this

Reply:
I didn't either so tried to improve it but am now deleting it. 😝